openverse_catalog/dags/providers/provider_api_scripts/rawpixel.py

```diff
@@ -1,193 +1,282 @@
-        # the dimensions in the metadata are at times
-        # inconsistent with the rescaled images
-        query_params = urlparse(img_url)
-        width = parse_qs(query_params.query).get("w", [])[0]
-        height = parse_qs(query_params.query).get("h", [])[0]
-        return [img_url, width, height]
-    else:
-        logger.warning(f"Image not detected in URL: {foreign_url}")
-        return [None, None, None, None]
-
-
-def _get_title_owner(image):
-    title = image.get("image_title", "")
-    owner = image.get("artist_names", "")
-    owner = owner.replace("(Source)", "").strip()
-    return [title, owner]
-
-
-def _get_meta_data(image):
-    description = image.get("pinterest_description")
-    meta_data = {}
-    if description:
-        meta_data["description"] = description
-    return meta_data
-
-
-def _get_tags(image):
-    keywords = image.get("keywords_raw")
-    if keywords:
-        keyword_list = keywords.split(",")
-        keyword_list = [
-            word.strip()
-            for word in keyword_list
-            if word.strip() not in ["cc0", "creative commons", "creative commons 0"]
-        ]
-        return keyword_list
-    else:
+    @staticmethod
+    def _get_image_url(data: dict, size_option: str) -> str | None:
+        """
+        Rawpixel provides a "style_uri" string which can be formatted with these values:
+            'image_24', 'image_png_24', 'image_48', 'image_png_48', 'image_100',
+            'image_png_100', 'image_150', 'image_png_150', 'image_200', 'image_png_200',
+            'image_250', 'image_png_250', 'image_300', 'image_png_300', 'image_400',
+            'image_png_400', 'image_500', 'image_png_500', 'image_600', 'image_png_600',
+            'image_700', 'image_png_700', 'image_800', 'image_png_800', 'image_900',
+            'image_png_900', 'image_1000', 'image_png_1000', 'image_1300',
+            'image_png_1300'
+
+        The number refers to the width displayed, and a png option is provided for each
+        size.
+        """
+        style_uri = data.get("style_uri")
+        if not style_uri:
+            return None
+        return style_uri.format(size_option)
+
+    @staticmethod
+    def _get_image_properties(data: dict) -> tuple[None, None] | tuple[int, int]:
+        width = max(data.get("width", 0), data.get("display_image_width", 0))
+        height = max(data.get("height", 0), data.get("display_image_height", 0))
+        if (width, height) == (0, 0):
+            return None, None
+        return width, height
+
+    @staticmethod
+    def _clean_text(text: str) -> str:
+        # Clean whitespace
+        text = text.strip()
+        # First clear full patterns
+        text = RawpixelDataIngester.suffix_pattern_full.sub("", text)
+        # Then clear partial patterns
+        text = RawpixelDataIngester.suffix_pattern_partial.sub("", text)
+        # Unescape HTMl sequences
+        text = html.unescape(text)
```

Comment:
This could be quite useful for other providers as well! (I think Jamendo is another one returning scaped text?)